service/telemetry/logger.go

```diff
@@ -28,9 +32,40 @@ func newLogger(cfg LogsConfig, options []zap.Option) (*zap.Logger, error) {
 	}
 
 	logger, err := zapCfg.Build(options...)
+
 	if err != nil {
 		return nil, err
 	}
+
+	if len(cfg.Processors) > 0 {
+		sdk, err := config.NewSDK(
+			config.WithContext(ctx),
+			config.WithOpenTelemetryConfiguration(
+				config.OpenTelemetryConfiguration{
+					LoggerProvider: &config.LoggerProvider{
+						Processors: cfg.Processors,
+					},
+					// TODO: add resource
+					// Resource: &config.Resource{},
+				},
+			),
+		)
+
+		if err != nil {
+			return nil, err
+		}
+
+		logger = logger.WithOptions(zap.WrapCore(func(c zapcore.Core) zapcore.Core {
+			return zapcore.NewTee(
```

Comment:
Using `NewTee` here means that logs will be written to both cores right? So logs would be written to stdout and exported? Are we sure we want to do that initially or is configuring a processor acknowledgement that logs won't go to stdout? For users that want the collector to stop writing logs to stdout and only use the bridge to export their logs they won't have a solution.

Reply:
i like the idea of having local logs as a fallback, which makes tee'ing nice in that sense. A user could configure the console log exporter if they wanted to have both local logs and data exported to another destination, but the output from the console will be significantly different.

Output from existing logger:
```
2024-10-22T15:14:38.081-0700    info    service@v0.111.0/service.go:280 Shutdown complete.
```

Output from console exporter:
```
{
        "Timestamp": "2024-10-22T15:14:38.081624-07:00",
        "ObservedTimestamp": "2024-10-22T15:14:38.08164-07:00",
        "Severity": 9,
        "SeverityText": "info",
        "Body": {
                "Type": "String",
                "Value": "Shutdown complete."
        },
        "Attributes": [],
        "TraceID": "00000000000000000000000000000000",
        "SpanID": "0000000000000000",
        "TraceFlags": "00",
        "Resource": [
                {
                        "Key": "service.name",
                        "Value": {
                                "Type": "STRING",
                                "Value": "otelcorecol"
                        }
                },
                {
                        "Key": "service.version",
                        "Value": {
                                "Type": "STRING",
                                "Value": "0.111.0-dev"
                        }
                },
                {
                        "Key": "telemetry.sdk.language",
                        "Value": {
                                "Type": "STRING",
                                "Value": "go"
                        }
                },
                {
                        "Key": "telemetry.sdk.name",
                        "Value": {
                                "Type": "STRING",
                                "Value": "opentelemetry"
                        }
                },
                {
                        "Key": "telemetry.sdk.version",
                        "Value": {
                                "Type": "STRING",
                                "Value": "1.31.0"
                        }
                }
        ],
        "Scope": {
                "Name": "go.opentelemetry.io/collector/service/telemetry",
                "Version": "",
                "SchemaURL": ""
        },
        "DroppedAttributes": 0
}
```

Reply:
One case where I think the fallback is interesting is if the OTLP logs exporting fails and we have some log from e.g. grpc-go that would be helpful in debugging

Reply:
This PR gives users the option to start producing logs over OTLP, without breaking existing functionality. I would suggest we follow up this PR with a feature gate that disables the tee'ing of the logs.

This would give users that don't want to double emit the logs to turn off the local console logging if a provider is configured. This could become the default in the future if users support this idea.

One thing to note is that configuration of the logging today supports sampling logs, which the otel logging provider has no concept of at this time

Reply:
That works for me. As long as we have some plan for how to give users the option to disable the console logging if they want I'm happy.